lib/puppet/indirector/request.rb

```diff
@@ -185,7 +185,8 @@ def do_request(srv_service=:puppet, default_server=nil, default_port=nil, &block
     return yield(self) if !self.server.nil?
 
     if Puppet.settings[:use_srv_records]
-      Puppet::Network::Resolver.each_srv_record(Puppet.settings[:srv_domain], srv_service) do |srv_server, srv_port|
+      resolver = Puppet.lookup(:dns_resolver)
```

Comment:
Could we create a new Resolver here instead of storing globally in the context? Similarly, create a new Resolver in the rest client's initialize method? This way for each agent run (every 30 minutes) we create a new client and its resolver (and don't need to worry about the Resolver cache growing unbounded for daemonized agents.

Reply:
We can create one in the rest client's initialize, but if we create a new one in this spot, we lose all caching, since this stuff happens per-request. I was going off your suggestion in https://github.com/puppetlabs/puppet/pull/5578 to use the context to store it, though I was probably going to move where it gets pushed on to the stack adjacent to where the http pool gets added here https://github.com/puppetlabs/puppet/blob/master/lib/puppet/configurer.rb#L211.

Reply:
Also, I am planning to add cache eviction for the record list based on its TTL, so that will prevent the unbounded growing.

Reply:
Ah right. I'm thinking since we don't cache SRV records now for indirector-based requests, I'd rather not implement that for legacy stuff.

Reply:
Alright. We can always start using it later once we've worked out the kinks, I suppose. I'll update.